solver_rspec.rb

```diff
@@ -0,0 +1,38 @@
+require_relative './solver'
+
+describe 'Test solver class' do
+  context 'Test factorial method' do
+    it 'factorial method with positive numer' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to be(120)
+    end
+    it 'factorial method with Negative numer' do
+      solver = Solver.new
+      expect { solver.factorial(-5) }.to raise_error('Error: Your number should be positive.')
+    end
+  end
```

Comment:
- [ ] For more test coverage, which is also a requirement for this project, please it will be nice to add a `special case - when input is 0, the result should equal 1`